src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.IO;
+using System.Text.Json;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using EnvDTE80;
+using Microsoft.VisualStudio.Shell.Interop;
+using Uno.UI.RemoteControl.VS.Helpers;
+using Microsoft.VisualStudio.PlatformUI;
+using Uno.UI.RemoteControl.VS.Notifications;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio;
+using System.Threading.Tasks;
+
+namespace Uno.UI.RemoteControl.VS;
+
+internal class GlobalJsonObserver
+{
+	private AsyncPackage _asyncPackage;
+	private DTE _dte;
+	private Action<string> _debugAction;
+	private readonly Action<string> _infoAction;
+	private readonly Action<string> _warningAction;
+	private readonly Action<string> _errorAction;
+	private FileSystemWatcher? _fileWatcher;
+	private readonly JsonSerializerOptions _readerOptions = new() { ReadCommentHandling = JsonCommentHandling.Skip };
+
+	public GlobalJsonObserver(
+		AsyncPackage asyncPackage
+		, DTE dte
+		, Action<string> debugAction
+		, Action<string> infoAction
+		, Action<string> warningAction
+		, Action<string> errorAction)
+	{
+		_asyncPackage = asyncPackage;
+		_dte = dte;
+		_debugAction = debugAction;
+		_infoAction = infoAction;
+		_warningAction = warningAction;
+		_errorAction = errorAction;
+
+		_debugAction($"GlobalJsonObserver: Starting");
+
+		try
+		{
+			Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
+			var globalJsonPath = GetGlobalJsonPath();
+
+			ObserveChanges(globalJsonPath);
+		}
+		catch (Exception e)
+		{
+			_debugAction($"GlobalJsonObserver: Failed to start " + e);
```

Comment:
```suggestion
			_debugAction($"GlobalJsonObserver: Failed to start - {e.Message}");
```

Reply:
The stack trace is important here, I'll keep as-is.

Reply:
oh yeah noted!